ecl/hql/hqlparse.cpp

```diff
@@ -568,6 +569,52 @@ void HqlLex::pushMacro(IHqlExpression *expr)
     }
 }
 
+void HqlLex::checkSignature()
+{
+    YYSTYPE dummyToken;
+    dummyToken.setPosition(0,0,0,text->querySourcePath());
+    Owned<IPipeProcess> pipe = createPipeProcess();
+    pipe->run("gpg", "gpg --verify -", ".", true, false, true, 0, false);
```

Comment:
It could be subverted but only if you have access to the eclcc machine as root or hpcc user – and if you have that you can disable signing anyway.